File: dotnet/cafdemalihapa/Applications/DirectoryManagement/DirectoryManagement.cs
namespace cafdemalihapa.Applications.DirectoryManagement
{
    public class DirectoryManagement
    (
        Creation creation,
        Opening opening,
        Shutting shutting
    )
    {
        public void Run()
        {
            if (!Applications.Get().IsDirectoryManagementApplication())
            {
                return;
            }
            opening.OpenDirectoryToBeOpen();
            creation.Create();
            opening.Open();
            shutting.Shut();
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/DirectoryManagement/DirectoryManagement.fs
namespace cafdemwimu.console.Applications.DirectoryManagement

open cafdemwimu.console.Applications

type DirectoryManagement(creation: Creation, opening: Opening, shutting: Shutting) =
    member _.Run() =
        if Applications.Get().IsDirectoryManagementApplication() then
            opening.OpenDirectoryToBeOpen()
            creation.Create()
            opening.Open()
            shutting.Shut()
